src/circuit.rs

```diff
@@ -74,10 +138,12 @@ where
             prover
                 .prover_key
                 .expect("Unexpected error. Missing ProverKey in compilation"),
-            verifier
-                .verifier_key
-                .expect("Unexpected error. Missing VerifierKey in compilation"),
-            pi_pos,
+            VerifierData::new(
+                verifier.verifier_key.expect(
```

Comment:
Guess the previous comment from @vlopes11 is also applying here.

Reply:
Is not. That is an `Option<T>`. I can transform it into an error. But again. if the Option is empty, that would be unexpected behaviour and therefore I consider panic a better way to handle it.